cclib/parser/data.py

```diff
@@ -290,6 +291,15 @@ def typecheck(self):
                 args = (attr, type(val), self._attributes[attr].type)
                 raise TypeError("attribute %s is %s instead of %s and could not be converted" % args)
 
+    def check_values(self, logger=logging):
+        """Perform custom checks on the values of attributes."""
+        if hasattr(self, "etenergies") and any(e < 0 for e in self.etenergies):
+            negative_values = [e for e in self.etenergies if e < 0]
+            msg = ("At least one excitation energy is negative. "
+                   "\nNegative values: %s\nFull etenergies: %s"
+                   % (negative_values, self.etenergies))
+            logger.error(msg)

```

Comment:
Using `ERROR` doesn't halt parsing, right?

Reply:
Not by default, no.